src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1854,8 +1831,9 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
                     // Convert elements to best type to determine element top-level nullability and to report nested nullability warnings
                     for (int i = 0; i < n; i++)
                     {
-                        var expression = expressions[i];
-                        resultTypes[i] = ApplyConversion(expression, expression, conversions[i], inferredType, resultTypes[i], checkConversion: true,
+                        var expressionNoConversion = expressionsNoConversions[i];
+                        var expression = GetConversionOperandIfApplicable(expressions[i], expressionNoConversion);
```

Comment:
>expression [](start = 28, length = 10)

nit: consider renaming `conversionOpt` to match the parameter name of `VisitConversion`